src/HotChocolate/AspNetCore/src/AspNetCore/Serialization/DefaultHttpRequestParser.cs

```diff
@@ -47,6 +51,88 @@ internal class DefaultHttpRequestParser : IHttpRequestParser
             CancellationToken cancellationToken) =>
             ReadAsync(stream, false, cancellationToken);
 
+        public async ValueTask<IReadOnlyList<GraphQLRequest>> ReadMultipartRequestAsync(
+            HttpRequest httpRequest,
+            CancellationToken cancellationToken)
+        {
+            // todo: The IFormCollection is convenient, but it requires us to work with strings instead of a stream
+            var formFeature = new FormFeature(httpRequest);
+            var form = await formFeature.ReadFormAsync(cancellationToken);
+
+            if (form.Count < 2)
+                throw new Exception("At least a 'operations' and 'map' field need to be present.");
+
+            if (!form.TryGetValue("operations", out StringValues operationsCollection) || !operationsCollection.TryPeek(out var operationsString))
+                throw new Exception("No 'operations' specified.");
+
+            if (!form.TryGetValue("map", out StringValues mapCollection) || !mapCollection.TryPeek(out var mapString))
+                throw new Exception("No 'map' specified.");
+
+            var map = JsonSerializer.Deserialize<IDictionary<string, string[]>>(mapString);
+
+            if (map == null)
+                throw new Exception("'map' is not a dictionary of string[].");
+
+            // todo: Stuff like this is just a temporary hack until we have a proper multipart reader
+            var operationsBytes = Encoding.UTF8.GetBytes(operationsString);
+            var operationsStream = new MemoryStream(operationsBytes);
+
+            var requests = await ReadAsync(operationsStream, false, cancellationToken);
```

Comment:
We should not use the ReadAsync method. Instead I will rework

```
public static unsafe IReadOnlyList<GraphQLRequest> Parse(
            string sourceText,
            ParserOptions? options = null)
```

on the underlying request parser to allow us to pass in the document cache.

Reply:
The read with the stream handles request size limits.... since the string is already in memory we do not need to bother about this anymore. That is why it actually would be better to just use the request parser ... I will do this so do not worry.